pkg/config/config.go

```diff
@@ -799,28 +799,67 @@ func ResolveVariables(
 		Variables: map[string]cty.Value{"var": cty.ObjectVal(origin)},
 	}
 	for key, val := range ctyMap {
-		if val.Type() == cty.String {
-			var valString string
-			if err := gocty.FromCtyValue(val, &valString); err != nil {
-				return err
+		newVal, err := resolveValue(val, evalCtx)
+		if err != nil {
+			return err
+		}
+		ctyMap[key] = newVal
+	}
+	return nil
+}
+
+func resolveValue(val cty.Value, evalCtx *hcl.EvalContext) (cty.Value, error) {
+	t := val.Type()
```

Comment:
You can use `cty.Transform` to automate "walk" and don't do "is it map?, is it list?"

Reply:
@nick-stroud this is a good suggestion so I'm removing your assignment for the moment.